workflows/energyplus.py

```diff
@@ -203,281 +190,306 @@ def run_energyplus(self, isIP, run_directory, file_name, args, by_api):
-                    for message in self.execute_for_callback(command_line_args, run_directory):
-                        self.callback(message)
-                except subprocess.CalledProcessError:
-                    self.callback("E+ FAILED")
-                    return EPLaunchWorkflowResponse1(
-                        success=False,
-                        message="EnergyPlus failed for file: %s!" % full_file_path,
-                        column_data={}
-                    )
+                expand_objects = os.path.join(energyplus_root_folder, 'ExpandObjects')
+            if not os.path.exists(expand_objects):
+                return EPLaunchWorkflowResponse1(
+                    success=False,
+                    message="ExpandObjects binary not found: {}!".format(expand_objects),
+                    column_data=[]
+                )
+            # go ahead and copy the target IDF and original IDD into the output_dir/in.idf for running ExpandObjects
+            idf_path_in_output_dir = os.path.join(output_directory, 'in.idf')
+            shutil.copy(full_file_path, idf_path_in_output_dir)
+            idd_path = os.path.join(energyplus_root_folder, 'Energy+.idd')  # yes for now we still need it
+            idd_path_in_output_dir = os.path.join(output_directory, 'Energy+.idd')
+            shutil.copy(idd_path, idd_path_in_output_dir)
+            # run ExpandObjects
+            try:
+                for message in self.execute_for_callback([expand_objects], output_directory):
+                    self.callback(message)
+            except subprocess.CalledProcessError:
+                self.callback("E+ FAILED")
+                return EPLaunchWorkflowResponse1(
+                    success=False,
+                    message="EnergyPlus failed for file: %s!" % full_file_path,
+                    column_data={}
+                )
+            # regardless of what happened, clean up the temp files
+            delete_if_exists(idf_path_in_output_dir)
+            delete_if_exists(idd_path_in_output_dir)
+            # if expanded.idf does not exist then we don't need to do anything, assume it just didn't have any templates
+            # but if it does exist, then this needs to be the IDF that we run, *however* we can't just leave it here
+            # as it may reference some other files back in the original directory.  I'm going to go with a new file name
+            # in the original directory of "FileName_Expanded.idf"
+            expanded_path_in_output_dir = os.path.join(output_directory, 'expanded.idf')
+            if os.path.exists(expanded_path_in_output_dir):
+                target_expanded_file_name = os.path.join(run_directory, f"{file_name_no_ext}_Expanded.idf")
+                shutil.copy(expanded_path_in_output_dir, target_expanded_file_name)
+                full_file_path = target_expanded_file_name
+
+        # Run EnergyPlus in a subdirectory
+        command_line_args += ['-d', output_directory]
```

Comment:
Here we ask EnergyPlus to run in the newly set up output directory, instead of running in the IDF directory. (pwd)